internal/http/proxy/proxy.go

```diff
@@ -30,6 +30,8 @@ func NewProxy(cfg Config, blockFieldSuggestions *block_field_suggestions.BlockFi
 	}
 	proxy := &httputil.ReverseProxy{
 		Rewrite: func(r *httputil.ProxyRequest) {
+			// Properly proxy XFF header as Rewrite removes it
+			r.Out.Header["X-Forwarded-For"] = r.In.Header["X-Forwarded-For"]
```

Comment:
Go recomments just calling r.SetXForwarded()
https://tip.golang.org/doc/go1.20#reverseproxy_rewrite
Line 111: https://go.dev/src/net/http/httputil/reverseproxy.go

Reply:
Updated to do both, by just calling `r.SetXForwarded()` the XFF is set to the client IP. By first copying the XFF from incoming over calling `r.SetXForwarded()` appends the clientIP to the XFF as according to the spec.